ctru-rs/src/applets/mii_selector.rs

```diff
@@ -0,0 +1,190 @@
+//! Mii Selector applet
+//!
+//! This module contains the methods to launch the Mii Selector.
+
+use crate::mii::MiiData;
+use bitflags::bitflags;
+use std::ffi::CString;
+
+/// Index of a Mii used to configure some parameters of the Mii Selector
+/// Can be either a single index, or _all_ Miis
+#[derive(Debug, Clone)]
+pub enum MiiConfigIndex {
+    Index(u32),
+    All,
+}
+
+/// The type of a Mii with their respective data
+#[derive(Debug, Clone)]
+pub enum MiiType {
+    Guest { index: u32, name: String },
+    User,
+}
+
+bitflags! {
+    /// Options for the Mii Selector
+    pub struct Options: u32 {
+        /// Show the cancel button
+        const MII_SELECTOR_CANCEL = ctru_sys::MIISELECTOR_CANCEL;
+        /// Make guest Miis selectable
+        const MII_SELECTOR_GUESTS = ctru_sys::MIISELECTOR_GUESTS;
+        /// Show on the top screen
+        const MII_SELECTOR_TOP = ctru_sys::MIISELECTOR_TOP;
+        /// Start on the guest's page
+        const MII_SELECTOR_GUEST_START = ctru_sys::MIISELECTOR_GUESTSTART;
+    }
+}
+
+/// An instance of the Mii Selector
+///
+/// # Example
+/// ```
+/// use ctru::applets::mii_selector::MiiSelector;
+///
+/// let mut mii_selector = MiiSelector::init();
+/// mii_selector.set_title("Example Mii selector");
+///
+/// let result = mii_selector.launch().unwrap();
+/// ```
+#[derive(Clone, Debug)]
+pub struct MiiSelector {
+    config: Box<ctru_sys::MiiSelectorConf>,
+}
+
+/// Return value from a MiiSelector's launch
+#[derive(Clone, Debug)]
+pub struct MiiSelectorReturn {
```

Comment:
`MiiSelector` and `MiiSelectorReturn` should be `non_exhaustive` to prevent the user from creating them without using the service.